mitodl/unified-ecommerce
# Reworks OpenAPI generation/setup to mirror other projects

### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/5723

### Description (What does it do?)

Updates the API routing and OpenAPI schema generation to mirror other projects (mainly Learn), so we can also produce a client build that'll end up in a separate repo. 

### How can this be tested?

Automated tests should pass and the app should function normally.

The schemas are now versioned - so instead of `/api/schema/` it's now `/api/v0/schema/`. The routes in APISIX have been updated for this change (so they continue to not require auth). 

This also adds in the GitHub actions to check the generated client for freshness. 

### Additional Context

Additional work is necessary to publish the resulting client. (For Learn, this is handled via Concourse so this will likely do the same.) However, the client needed to be built properly first and things weren't versioned correctly - the APIs we have now are all `v0` but the resulting client wasn't versioned. 

Some more config will be needed when the APIs move to v1/v2/etc. (as expected). 

This won't pass checks as is - there shouldn't be any outright errors but there will be some warnings and the check is set up to fail on warnings. A lot of these are in the payments APIs (specifically the basket part of that). We have another issue open to do some work there, so we should fix it in that PR instead.
